source/common/stats/stats_impl.h

```diff
@@ -366,9 +396,135 @@ class HistogramImpl : public Histogram, public MetricImpl {
   // Stats::Histogram
   void recordValue(uint64_t value) override { parent_.deliverHistogramToSinks(*this, value); }
 
+  void merge() override {}
+
+  bool used() const override { return true; }
+
+  HistogramStatistics intervalStatistics() const override { return HistogramStatistics(); }
+
+  HistogramStatistics cumulativeStatistics() const override { return HistogramStatistics(); }
+
   Store& parent_;
 };
 
+/**
+ * Log Linear Histogram implementation per thread.
```

Comment:
Can we move  ThreadLocalHistogramImpl and HistogramParentImpl directly into thread_local_store. We are going to have to deal with some locking issues and I would like things to be more explicitly coupled. We can break it apart later if necessary.